fw/compile-firmware.md

```diff
@@ -0,0 +1,146 @@
+---
+layout: default
+title: Compiling your firmware
+nav_order: 99
+parent: Firmware
+---
+
+# Table of contents
+
+1. TOC
+{:toc}
+
+# Introduction
+
+The [BastardKB QMK repository](https://github.com/bastardkb/bastardkb-qmk) contains the firmware code and releases for the Bastard Keyboards Charybdis boards.
+
+While that repository is used as primary source of truth for Bastard Keyboards firmwares as it contains the latest changes and improvements, the maintainers aim at upstreaming all those changes to the official [QMK repository](https://github.com/qmk/qmk_firmware).
+
+While the behavioural code is stored on the QMK repository, the keymaps are stored in our [QMK userspace](https://github.com/Bastardkb/qmk_userspace).
+
+This page details how to compile your own firmware. Building from source is useful to people who want to customize their keyboard and keymaps beyond what Via offers. This involves using a command-line interface, writing C code, and compiling this code into the final firmware file.
+
+If that seems too complicated, you can also use one of the default releases, or VIA.
+
+# Pre-requisites
+
+## Working QMK environment
+
+Make sure you have a functional QMK environment. See [QMK Docs](https://docs.qmk.fm/#/newbs) for details. At this point, **you don't need** to run `qmk setup`.
+
+## BastardKb QMK fork
+
+Clone the BKB QMK repository, using either github desktop or the command line, and switch to the `bkb-master` branch:
+
+```shell
+git clone https://github.com/bastardkb/bastardkb-qmk
+cd bastardkb-qmk
+git checkout origin/bkb-master
+```
+
+Now that you've cloned the repository, `cd` into it and set it as the default local QMK repository. You also need to do this if you had a previous, separate QMK installation.
+
+```shell
+qmk config user.qmk_home="$(realpath .)" 
+```
+
+{: .info }
+If you have multiple QMK installations, you will need to manually set the qmk home path again to use the other ones.
+
+## BastardKb userspace
+
+While the QMK repository contains the logic behind the keyboards, the keymaps are in the userspace repository.
+
+In a separate folder, clone the BKB QMK repository, using either github desktop or the command line:
+
+
+```shell
+git clone https://github.com/bastardkb/qmk_userspace
+```
+
+Next, `cd` into the repository and enable userspace:
+
+```shell
+cd qmk_userspace
+qmk config user.overlay_dir="$(realpath .)"
+```
+
+## Github actions
+
+// TODO screenshot enable actions workflows
+
+// TODO mroe details, screenshots
```

Comment:
```suggestion
// TODO more details, screenshots
```

Reply:
not relevant, as TODO's will be removed hopefully
I admire the dedication though